src/App.tsx

```diff
@@ -1,16 +1,143 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useContext,
+  useCallback,
+  useEffect,
+  useState,
+} from 'react';
+import { Todo } from './types/Todo';
+import { User } from './types/User';
+import { Error } from './types/Error';
 import { AuthContext } from './components/Auth/AuthContext';
+import {
+  getTodos,
+  addTodo,
+  deleteTodo,
+  changeTodo,
+} from './api/todos';
+
+import { TodoList } from './components/TodoList/TodoList';
+import { NewTodo } from './components/NewTodo/NewTodo';
+import { TodoError } from './components/TodoError/TodoError';
 
 export const App: React.FC = () => {
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
-  const newTodoField = useRef<HTMLInputElement>(null);
+  const [userTodos, setUserTodos] = useState<Todo[]>([]);
+  const [hasTodos, setHasTodos] = useState(false);
+  const [hasError, setHasError] = useState<Error>({ status: false });
```

Comment:
from `hasError` name I expect that this state will have boolean value, but instead there is an Error object

use more clear names, in this case errorState will be more clear